mantidproject/mantid
# Tidy Code & Functionality in the Reflectometry GUI

**Description of work.**
- Increaced the size of the Runs column by default.
- Switched hard coded indecies for use of the `Column` enum for maintainability.
- Removed uses of the presenter member variable in `MainWindowView`
- Globally defined colours used in the GUI tables
- Removed unused methods from `InstrumentView`
- Moved implementation out of `JobsViewUpdater` header file
- Fixed validation of ExperimentView when loading batches
- Disabled Add Row button in Experiment tab when processing
- Improved naming of notification functions in Save, Event, Experiment, and Instrument components
- Removed uses of `RowPath`, replacing them with `RowLocation`

**To test:**
 I know it's a cop-out to not write proper testing instructions, but given so many parts of the GUI code have been touched it's probably worth just going through the manual testing: https://developer.mantidproject.org/Testing/ReflectometryGUI/ReflectometryGUITests.html

Otherwise: 
1. Check that on opening the GUI the Runs column is wide enough that you can click inside the *second* row. 
2. Check that when processing runs in the runs tab the Add Row button is disabled in the Experiment Settings tab. 
3. Put some invalid values into the Experiment Settings -> Per-angle defaults table and save the batch. Reloading the batch should still show these invalid values as errors. 

Fixes #26539 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
